burn-compute/tests/integration_test.rs

```diff
@@ -146,3 +157,74 @@ fn autotune_cache_miss_test() {
     // Cache should be missed, so CacheTestSlowOn3 (but faster on 5) should be used, returning rhs
     assert_eq!(obtained_resource.read(), Vec::from([5, 6, 7, 8, 9]));
 }
+
+#[test]
+#[serial]
+#[cfg(feature = "std")]
+fn autotune_cache_different_keys_return_a_cache_miss() {
+    let client = client(&DummyDevice);
+
+    // in this test shapes [1,3] and [1,5] ends up with different key names
+    // which are 'cache_test-1,4' and 'cache_test-1,8'
+    let shapes_1 = vec![vec![1, 3], vec![1, 3], vec![1, 3]];
+    let lhs_1 = client.create(&[0, 1, 2]);
+    let rhs_1 = client.create(&[4, 4, 4]);
+    let out_1 = client.empty(3);
+    let handles_1 = vec![lhs_1, rhs_1, out_1];
+
+    let shapes_2 = vec![vec![1, 5], vec![1, 5], vec![1, 5]];
+    let lhs_2 = client.create(&[0, 1, 2, 3, 4]);
+    let rhs_2 = client.create(&[5, 6, 7, 8, 9]);
+    let out_2 = client.empty(5);
+    let handles_2 = vec![lhs_2, rhs_2, out_2.clone()];
+
+    let cache_test_autotune_kernel_1 =
+        dummy::CacheTestAutotuneOperationSet::new(client.clone(), shapes_1, handles_1);
+    let cache_test_autotune_kernel_2 =
+        dummy::CacheTestAutotuneOperationSet::new(client.clone(), shapes_2, handles_2);
+    client.execute_autotune(Box::new(cache_test_autotune_kernel_1));
+    client.execute_autotune(Box::new(cache_test_autotune_kernel_2));
+
+    let obtained_resource = client.read(&out_2);
+
+    // Cache should be missed, so CacheTestSlowOn3 (but faster on 5) should be used, returning rhs
+    assert_eq!(obtained_resource.read(), Vec::from([5, 6, 7, 8, 9]));
+}
+
+#[test]
+#[serial]
+#[cfg(feature = "std")]
+fn autotune_cache_different_checksums_return_a_cache_miss() {
+    // we use two clients in order to have freshly initialized autotune caches
+    let client1 = client(&DummyDevice);
+    let client2 = client2(&DummyDevice);
```

Comment:
I think the second client should be generated after the cache is saved on disk. For that, I think we need to create a new client (not just a client to a static compute).